### ISSUE TITLE
Descent: Anniversary Edition crash

### ISSUE THREAD

**Issue by mattcaron** [MiSTer-devel/ao486_MiSTer#39]
Build: `ao486_20200907.rbf`

Reproduce: Run descent anniversary edition.

Dump: See attached screenshot.

![20200907_224112-screen](https://user-images.githubusercontent.com/487889/92419232-35216680-f13a-11ea-813f-3bbf578c80db.png)

**Comment by mattcaron** [comment]
Might be same as https://github.com/MiSTer-devel/ao486_MiSTer/issues/40

**Comment by ChrisRaynor** [comment]
Too much memory for DOS extender, either patch it or reduce your system memory right down to something more realistic (16 MB)

**Comment by mattcaron** [comment]
When I filed this, it was already set to 16MB.

I will retest with an up to date build.

**Comment by ChrisRaynor** [comment]
Try this...

https://misterfpga.org/viewtopic.php?t=951

**Comment by mattcaron** [comment]
I had found that, but had resisted using that approach because isn't the point of MiSTer to be faithful reproductions of original things? Therefore, I figured I'd report the bug and if it necessitated some sort of additional option (say, a 4MB or 8MB RAM menu option), then that would be a valid fix and would to not require running it inside a wrapper or replacing any files.

Or do I misunderstand the intent of the project?

**Comment by ChrisRaynor** [comment]
Not sure but a lower memory option might come in handy I guess..

**Comment by mattcaron** [comment]
Okay, retested this with the 20210117 release.

1. It still crashes with a similar error to the above (I did not do a line by line compare to see if they were identical, but they are broadly the same).
2. Running it under DOS32A as described in https://misterfpga.org/viewtopic.php?t=951 does provide a workaround.

But, as we said before, a workaround should not be necessary.

Indeed, if it is as simple as "too much memory", I would wonder how hard it is to make an 8MB RAM option and just use that for a generic dos 6.22 box. I would think that more memory would only be useful if one is using it to run Win95.

**Comment by wishusknight** [comment]
> 
> 
> Okay, retested this with the 20210117 release.
> 
>     1. It still crashes with a similar error to the above (I did not do a line by line compare to see if they were identical, but they are broadly the same).
> 
>     2. Running it under DOS32A as described in https://misterfpga.org/viewtopic.php?t=951 does provide a workaround.
> 
> 
> But, as we said before, a workaround should not be necessary.
> 
> Indeed, if it is as simple as "too much memory", I would wonder how hard it is to make an 8MB RAM option and just use that for a generic dos 6.22 box. I would think that more memory would only be useful if one is using it to run Win95.

So far as I understand it, isn't so much how much memory is selected in the menu, it is the method used to limit memory in AO486 that doesn't play well with very specific scenario's. To change this behavior necessitates a pretty large amount of work that would probably break more than it fixes. And the specific scenario is originally a bug in the dos extender that had a fix for it in a later release. That same bug manifests itself in newer real hardware as well that uses artificial bios memory limits. So in a sense AO486 is working as intended.

**Comment by mattcaron** [comment]
If your assessment is accurate (and the test of the theory would seem to confirm this), then I agree with your conclusion. I just find it surprising for a few reasons:

1. This isn't the originally released Descent, it is the re-released anniversary edition from 1996, which, one would think, have been patched.
2. Especially since Doom was released earlier and runs fine under AO486. (Though I suppose that the bug could have been introduced in a later extender version and Descent is in that window of bugginess)
3. In the late 90's I was running a PII-450 and with 128MB RAM, and I would swear that this game worked. This was under Win95's dos when you could have it shut down and drop you at a DOS prompt, so would have been DOS 7. I wasn't limiting the amount of memory at all in the BIOS. Now, that said, this was also 20 years ago, so it could just be my bad memory.

I wonder if Descent II is similarly broken. I'll have to test that.

Anyway, in conclusion, I'd say that this can probably be closed. There is a solid theory as to why it is happening (so it's not a random unknown) and we have a workaround for it which I have confirmed works. As such, I will close this.

**Comment by mattcaron** [comment]
There is an extensive discussion about this here: https://github.com/MiSTer-devel/ao486_MiSTer/issues/45
